orbit/pkg/keystore/keystore_darwin.go

```diff
@@ -0,0 +1,158 @@
+//go:build darwin && cgo
+
+package keystore
+
+/*
+   #cgo LDFLAGS: -framework CoreFoundation -framework Security
+   #include <CoreFoundation/CoreFoundation.h>
+   #include <Security/Security.h>
+*/
+import "C"
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+)
+
+const service = "com.fleetdm.fleetd.enroll.secret"
+
+var serviceStringRef = stringToCFString(service)
+
+func Exists() bool {
+	return true
+}
+
+func Name() string {
+	return "default keychain"
+}
+
+// AddSecret will add a secret to the keychain. This secret can be retrieved by this application without any user authorization.
+func AddSecret(secret string) error {
```

Comment:
Out of curiosity: What happens if one executes this method twice? (It adds two secrets?, which one is valid?)

Reply:
It will fail for macOS keychain. We need to call updateSecret instead.